src/ACL.cpp

```diff
@@ -139,38 +139,80 @@ QFlags< ChanACL::Perm > ChanACL::effectivePermissions(ServerUser *p, Channel *ch
 
 	bool traverse = true;
 	bool write    = false;
-	ChanACL *acl;
 
+	// Iterate over all parent channels from root to the channel the user is in (inclusive)
 	while (!chanstack.isEmpty()) {
 		ch = chanstack.pop();
-		if (!ch->bInheritACL)
+		if (!ch->bInheritACL) {
 			granted = def;
+		}
 
-		foreach (acl, ch->qlACL) {
+		for (ChanACL *acl : ch->qlACL) {
 			bool matchUser  = (acl->iUserId != -1) && (acl->iUserId == p->iId);
 			bool matchGroup = Group::appliesToUser(*chan, *ch, acl->qsGroup, *p);
+
+			bool applyFromSelf   = (ch == chan && acl->bApplyHere);
+			bool applyFromParent = (ch != chan && acl->bApplySubs);
```

Comment:
```suggestion
			bool applyInherited = (ch != chan && acl->bApplySubs);
```
`parent` seems to suggest that the inheritance comes from the immediate parent (which technically is true but I think for a better understanding this detail is more confusing that helpful)